modules/standard/IO.chpl

```diff
@@ -7959,7 +7959,7 @@ proc fileReader.readBinary(ref data: [?d] ?t, param endian = ioendian.native): b
                        due to a :ref:`system error<io-general-sys-error>`.
 */
 proc fileReader.readBinary(ref data: [?d] ?t, param endian = ioendian.native): int throws
-  where ReadBinaryArrayReturnInt == true && (d.rank == 1 && d.stridable == false) && (
+  where ReadBinaryArrayReturnInt == true && (d.rank == 1 && d.stridable == false && !d.isSparse()) && (
```

Comment:
Can we still call readBinary with sparse domains?  If not, it would be good to call that out in the PR message as well.

Reply:
That's a good question. @mppf had me make the same change to `writeBinary` on another recent PR, so I did the same here.

Removing that component from the where-clause, and running the following program (with `-sReadBinaryArrayReturnInt=true`):
```chapel
use IO;

const x : [1..10] int = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
      fw = openWriter("test.bin");
fw.writeBinary(x);
fw.close();

const fr = openReader("test.bin");
var D = {1..10},
    d : sparse subdomain(D) = {1, 3, 5, 7, 9},
    y : [d] int;

writeln(fr.readBinary(y));
writeln(y);
```
I get the output:
```
5
1 2 3 4 5
```

Would we actually want the following instead? :
```
5
1 3 5 7 9
```

Reply:
I'm not actually sure.  It makes sense in that it is reading the data it was given and putting that in all the spots, but I could see wanting to skip to match the data as well.  Maybe make a separate version that is marked unstable but does the current behavior and open an issue to discuss it?  I think that would be okay as a post-2.0 thing to decide because iirc other sparse stuff is unstable, but others might disagree

Reply:
Okay sounds good. New issue here: https://github.com/chapel-lang/chapel/issues/22076